### PR TITLE
[MM-42713] - Remove cloud user limit restrictions

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR removes cloud user limitation number from the webapp

#### Ticket Link
[MM-42713](https://mattermost.atlassian.net/browse/MM-42713)

#### Related Pull Requests

- Has server changes (https://github.com/mattermost/mattermost-server/pull/19835)

#### Release Note

```release-note
Remove cloud user limit count
```

### CODE REVIEW

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx [hunk 1]
```diff
@@ -28,7 +28,6 @@ import {t} from 'utils/i18n';
 import AnnouncementBar from '../default_announcement_bar';
```
Comment (AGMETEOR) [inline components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx:28]: Should probably get rid of this entire file

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx [hunk 2]
```diff
@@ -28,7 +28,6 @@ import {t} from 'utils/i18n';
...
 
 type Props = {
```
Comment (pvev) [inline components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx:30]: if we remove this logic from here, is there any reason to keep these files? I think the mere purpouse of this announcement bar is notify about the user limit, so I guess we can get rid of them.

File: components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx [hunk 3]
```diff
@@ -149,12 +137,6 @@ class UserLimitAnnouncementBar extends React.PureComponent<Props> {
             return null;
         }
 
-        // If AT user limit, and banner hidden, don't render anything
-        if (userLimit === analytics!.TOTAL_USERS &&
-            preferences.some((pref) => pref.name === CloudBanners.HIDE && pref.value === 'true')) {
-            return null;
-        }
-
         const dismissable = this.isDismissable();
```
Comment (neallred) [inline components/announcement_bar/cloud_announcement_bar/user_limit_announcement_bar.tsx:140]: Nit: can remove `isDismissable` method, and have the `message=` down below always be `t('upgrade.cloud_banner_reached')`

File: components/post_view/channel_intro_message/index.ts [hunk 4]
```diff
@@ -35,10 +35,7 @@ function mapStateToProps(state: GlobalState) {
     const creator = getUser(state, channel.creator_id);
     const boardComponent = getChannelIntroPluginComponents(state).find((c) => c.pluginId === 'focalboard');
 
-    let usersLimit = parseInt(getConfig(state).ExperimentalCloudUserLimit! || '10', 10);
-    if (usersLimit === 0) {
-        usersLimit = 10;
-    }
+    const usersLimit = 0;
```
Comment (pvev) [inline components/post_view/channel_intro_message/index.ts:38]: this userLimit value was attached with the cloudUserLimit for initial requirements as a fallback, but now that we are getting rid off the cloudUserLimit value, this value must actually be set to 10 as stated in the original ticket https://mattermost.atlassian.net/browse/MM-32642. This is used to determine a minimum number of users existing in an environment and determine if you must invite members (if there is already minimum 10 existing users in the team), or invite people to join the team (if there are less than 10 users in the team). @marianunez just to confirm, should we keep 10 as the threshold number?

Comment (pvev) [reply components/post_view/channel_intro_message/index.ts:38]: also you should remove this call from here `e2e/cypress/tests/integration/accessibility/accessibility_buttons_spec.js` line 19

**Review by pvev — CHANGES_REQUESTED** [review]
Great work @AGMETEOR . Just two comments that need to be addressed. I will approve as soon as you attend those. Please, request re-review once this gets done. Thanks.

File: components/post_view/channel_intro_message/index.ts (hunk 4 above)
Comment (marianunez) [reply components/post_view/channel_intro_message/index.ts:38]: > This is used to determine a minimum number of users existing in an environment and determine if you must invite members (if there is already minimum 10 existing users in the team), or invite people to join the team (if there are less than 10 users in the team).

This is correct, we were pigginback on the user limit for but in reality we are not restricting anything. As Pablo mentioned, 10 was just a number to show a new server and promote the `Invite Members` button in the channel intro message area. 

We should just default to 0. Maybe change the variable name to make it more understandable or add a comment for future reference would be nice

File: components/widgets/menu/menu_items/menu_cloud_trial.test.tsx [hunk 5]
```diff
@@ -37,28 +37,6 @@ describe('components/widgets/menu/menu_items/menu_cloud_trial', () => {
         expect(wrapper.find('UpgradeLink').exists()).toEqual(true);
     });
 
-    test('should NOT render when NOT on cloud license and NOT during free trial period', () => {
```
Comment (marianunez) [inline components/widgets/menu/menu_items/menu_cloud_trial.test.tsx:40]: Is this behavior related to the user limit? It seems that it's testing if the Upgrade to trial link is available or not? We may want to keep this test

**Review by marianunez — COMMENTED** [review]
Awesome clean up work @AGMETEOR! 🎉

File: components/admin_console/billing/plan_details/plan_details.tsx [hunk 6]
```diff
@@ -263,13 +246,6 @@ export const getPlanPricing = (
                         defaultMessage='Free'
                     />
                 </div>
-                <div className='PlanDetails__planCaveat'>
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/admin_console/billing/plan_details/plan_details.tsx:266]: Removed

File: components/post_view/channel_intro_message/index.ts (hunk 4 above)
Comment (pvev) [reply components/post_view/channel_intro_message/index.ts:38]: @marianunez @AGMETEOR  I think the number should be keep as 10 (or any other number we/PM define). This number determines if, let's say we keep it as 10, if less than 10 users we show the "invite people to team", if more than 10, we show the "add members to channel".

Captura de pantalla 2022-03-29 a las 9 15 32
Captura de pantalla 2022-03-29 a las 9 16 26

Feel free to rename variables as necessary because the variable names are confusing.

**Review by pvev — APPROVED** [review]
Awesome work @AGMETEOR. Approving now cause the only thing left is define which number value to set as stated in the PR comments.

**Review by neallred — APPROVED** [review]
LGTM, thanks @AGMETEOR !

File: components/post_view/channel_intro_message/index.ts (hunk 4 above)
Comment (marianunez) [reply components/post_view/channel_intro_message/index.ts]: Yes, meant default to 10 - oops typo on my side. Thanks for the detailed clarification here!

**Review by marianunez — APPROVED** [review]

**Comment by marianunez** [comment]
@AGMETEOR reminder to double check if there is any cypress tests that might be affected by this refactor and that we are not breaking any tests with this PR
